webapp/platform/client/src/websocket.ts

```diff
@@ -156,6 +163,32 @@ export default class WebSocketClient {
                 this.firstConnectListeners.forEach((listener) => listener());
             }
 
+            if (this.config.clientPingEnabled) {
+                var waitingForPong = false;
+                this.pingInterval = setInterval(
+                    () => {
+                        if (!waitingForPong) {
+                            waitingForPong = true;
+                            this.ping(() => {
+                                waitingForPong = false;
+                            });
+                            return;
+                        }
+
+                        console.log('ping received no response within time limit: re-establishing websocket'); //eslint-disable-line no-console
+
+                        // We are not calling this.close() because we need to auto-restart.
+                        this.connectFailCount = 0;
+                        this.responseSequence = 1;
+                        if (this.pingInterval) {
+                            clearInterval(this.pingInterval);
+                            this.pingInterval = null;
+                        }
```

Comment:
These 4 lines are being repeated thrice. Refactor?

Reply:
Good idea - extracted this into a function.